vm.c

```diff
@@ -2132,6 +2132,14 @@ rb_vm_check_redefinition_opt_method(const rb_method_entry_t *me, VALUE klass)
         if (st_lookup(vm_opt_method_def_table, (st_data_t)me->def, &bop)) {
             int flag = vm_redefinition_check_flag(klass);
             if (flag != 0) {
+                if (rb_warning_category_enabled_p(RB_WARN_CATEGORY_PERFORMANCE)) {
+                    rb_category_warn(
+                        RB_WARN_CATEGORY_PERFORMANCE,
+                        "Redefining '%s#%s' disables interpreter and JIT optimizations",
+                        rb_class2name(me->owner),
+                        rb_id2name(me->called_id)
+                    );
+                }
```

Comment:
Checking `rb_warning_category_enabled_p` looks redundant.

```suggestion
                rb_category_warn(
                    RB_WARN_CATEGORY_PERFORMANCE,
                    "Redefining '%s#%s' disables interpreter and JIT optimizations",
                    me->owner, ID2SYM(me->called_id));
```